widget/list.go

```diff
@@ -60,22 +71,43 @@ func (l *List) MinSize() fyne.Size {
 	return l.BaseWidget.MinSize()
 }
 
+// SetSelected sets the index of the data that is to be selected, refreshes, and calls the OnSelected callback.
+func (l *List) SetSelected(index int) {
+	length := 0
+	if f := l.Length; f != nil {
+		length = f()
+	}
+	if index < 0 || index >= length {
+		return
+	}
+	l.selectedIndex = index
+	if index >= l.firstItemIndex && index <= l.lastItemIndex {
+		l.Refresh()
+		return
+	}
+	if index < l.firstItemIndex {
+		l.scroller.Offset.Y = index * l.itemMin.Height
+	} else {
+		l.scroller.Offset.Y = (index * l.itemMin.Height) - ((l.visibleItemCount - 1) * l.itemMin.Height)
```

Comment:
I can't quite figure out this maths.
Shouldn't it just be ((index+1)*minheight) - scrollerHeight?

Reply:
Yes good call

Reply:
Resolved